### PR TITLE
doc: Update FWU with external memory docs

### PR DESCRIPTION
**Pull request by tomchy**
Update SUIT documentation about firmware updates with external memory to match with the latest NCS codebase.

Ref: NCSDK-29735

### CODE REVIEW

**Review by maciejpietras — APPROVED** [review]

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst [hunk 1]
```diff
@@ -195,6 +248,78 @@ Steps specific for the fetch model
 
       This snippet assumes that ``CACHE_POOL`` is the third component on the manifest's components list (so its component index is 2)
 
+#. If your application uses radio core, modify the radio manifest file :file:`rad_envelope.yaml.jinja2` by completing the following steps:
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst:251]:
```suggestion
#. If your application uses the radio core, modify the radio core manifest file :file:`rad_envelope.yaml.jinja2` by completing the following steps:
```

**Review by FrancescoSer — APPROVED** [review]
Approving but please implement this small comment

File: samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2 [hunk 2]
```diff
@@ -239,20 +249,30 @@ SUIT_Envelope_Tagged:
...
-            envelope: {{ artifacts_folder ~ radio['name'] }}.suit
+            envelope: {{ artifacts_folder ~ application['name'] }}.suit
     - suit-directive-fetch:
       - suit-send-record-failure
-    - suit-condition-image-match:
-      - suit-send-record-success
-      - suit-send-record-failure
-      - suit-send-sysinfo-success
-      - suit-send-sysinfo-failure
+    # In the case of streaming operations it is worth to retry them at least once.
```
Comment (ahasztag) [inline samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2:261]: As discussed, this change seems a bit pointless for the root manifest as there actually is no streaming - just acquiring the digest fom a dependency manifest, which has to be in the MRAM memory anyway.
It was changed in the normal root manifest, but  I will change it back in an upcoming PR.

File: samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2 [hunk 3]
```diff
@@ -273,11 +293,21 @@ SUIT_Envelope_Tagged:
             envelope: {{ sysbuild['config']['SB_CONFIG_SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY'] }}/nordic_top.suit
     - suit-directive-fetch:
       - suit-send-record-failure
-    - suit-condition-image-match:
-      - suit-send-record-success
-      - suit-send-record-failure
-      - suit-send-sysinfo-success
-      - suit-send-sysinfo-failure
+    # In the case of streaming operations it is worth to retry them at least once.
```
Comment (ahasztag) [inline samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2:296]: Same case as before

**Review by ahasztag — CHANGES_REQUESTED** [review]

File: samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2 (hunk 3 above)
Comment (tomchy) [reply samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2]: Removed!

File: samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2 (hunk 2 above)
Comment (tomchy) [reply samples/suit/smp_transfer/suit/nrf54h20/root_with_binary_nordic_top_extflash.yaml.jinja2]: Removed!

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst [hunk 4]
```diff
@@ -137,16 +137,69 @@ Common steps for both push and fetch models
...
+           - suit-send-record-success
+           - suit-send-record-failure
+           - suit-send-sysinfo-success
+           - suit-send-sysinfo-failure
+         # Boot the companion application.
          - suit-directive-invoke:
-            - suit-send-record-failure
+           - suit-send-record-failure
+
+      The update procedure robustness may be further improved by copying all integrity checks from the ``suit-install`` to the ``suit-candidate-verification`` sequence.
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst:189]:
```suggestion
      To further improve the update procedure robustness, you can copy all integrity checks from the ``suit-install`` to the ``suit-candidate-verification`` sequence.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst [hunk 5]
```diff
@@ -137,16 +137,69 @@ Common steps for both push and fetch models
...
 
-      The companion image can be optionally upgraded and have its integrity checked.
+   #. Append flash companion binary as integrated payload inside SUIT envelope:
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst:191]:
```suggestion
   #. Append the flash companion binary as integrated payload inside SUIT envelope:
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst [hunk 6]
```diff
@@ -195,6 +248,78 @@ Steps specific for the fetch model
 
       This snippet assumes that ``CACHE_POOL`` is the third component on the manifest's components list (so its component index is 2)
 
+#. If your application uses the radio core, modify the radio core manifest file :file:`rad_envelope.yaml.jinja2` by completing the following steps:
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst:251]:
```suggestion
#. If your application uses the radio core, complete the following steps to modify the radio core manifest file :file:`rad_envelope.yaml.jinja2`:
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst [hunk 7]
```diff
@@ -195,6 +248,78 @@ Steps specific for the fetch model
...
+      .. code-block:: yaml
+
+         suit-payload-fetch:
+         - suit-directive-set-component-index: 2
+         - suit-directive-override-parameters:
+             suit-parameter-uri: 'file://{{ radio['binary'] }}'
+         - suit-directive-fetch:
+           - suit-send-record-failure
+
+      This snippet assumes that ``CACHE_POOL`` is the third component on the manifest's components list (so its component index is 2)
```
Comment (peknis) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_external_memory.rst:275]:
```suggestion
      This snippet assumes that ``CACHE_POOL`` is the third component on the manifest's components list (so its component index is ``2``)
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Comment by SylwesterKonczyk** [comment]
Looking at ug_nrf54h20_suit_external_memory.rst, line 62: "In the fetch model, the SUIT processor runs on the application core." Actually - Ability of pushing raw cache images to cache partitions and presence of SUIT processor are orthogonal. To give an example:

It is nothing wrong with the system, where SUIT processor is active on App core, and, at the same time **some** of images are 'pre loaded' to cache partition using 'push' mode. Once candidate manifests are being processed, other images may be fetched and stored in the same **cache** partition.

**Review by SylwesterKonczyk — APPROVED** [review]

**Review by ahasztag — APPROVED** [review]

**Pull request merged**